internal/services/mobilenetwork/mobile_network_packet_core_control_plane_resource.go

```diff
@@ -0,0 +1,713 @@
+			}
+
+			if metadata.ResourceData.HasChange("core_network_technology") {
+				value := packetcorecontrolplane.CoreNetworkType(plan.CoreNetworkTechnology)
+				model.Properties.CoreNetworkTechnology = &value
+			}
+
+			if metadata.ResourceData.HasChange("interoperability_settings_json") {
+				var interopSettingsValue interface{}
+				err := json.Unmarshal([]byte(plan.InteropSettings), &interopSettingsValue)
+				if err != nil {
+					return err
+				}
+
+				model.Properties.InteropSettings = &interopSettingsValue
+			}
+
+			if metadata.ResourceData.HasChange("local_diagnostics_access") {
+				model.Properties.LocalDiagnosticsAccess = expandPacketCoreControlLocalDiagnosticsAccessConfiguration(plan.LocalDiagnosticsAccess)
+			}
+
+			if metadata.ResourceData.HasChange("mobile_network_id") {
+				model.Properties.Sites = expandPacketCoreControlPlaneSites(plan.SiteIds)
+			}
+
+			if metadata.ResourceData.HasChange("platform") {
+				model.Properties.Platform, err = expandPlatformConfigurationModel(plan.Platform)
+				if err != nil {
+					return err
+				}
+			}
+
+			if metadata.ResourceData.HasChange("sku") {
+				model.Properties.Sku = packetcorecontrolplane.BillingSku(plan.Sku)
+			}
+
+			if metadata.ResourceData.HasChange("version") && plan.Version != "" {
+				model.Properties.Version = &plan.Version
+			}
+
+			if metadata.ResourceData.HasChange("tags") {
+				model.Tags = &plan.Tags
+			}
+
+			if err := client.CreateOrUpdateThenPoll(ctx, *id, model); err != nil {
+				return fmt.Errorf("updating %s: %+v", *id, err)
+			}
+
+			return nil
+		},
+	}
+}
+
+func (r PacketCoreControlPlaneResource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.MobileNetwork.PacketCoreControlPlaneClient
+
+			id, err := packetcorecontrolplane.ParsePacketCoreControlPlaneID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			resp, err := client.Get(ctx, *id)
+			if err != nil {
+				if response.WasNotFound(resp.HttpResponse) {
+					return metadata.MarkAsGone(id)
+				}
+
+				return fmt.Errorf("retrieving %s: %+v", *id, err)
+			}
+
+			state := PacketCoreControlPlaneModel{
+				Name:              id.PacketCoreControlPlaneName,
+				ResourceGroupName: id.ResourceGroupName,
+			}
+
+			if model := resp.Model; model != nil {
+				state.Location = location.Normalize(model.Location)
+
+				if model.Properties.UeMtu != nil {
+					state.UeMtu = *model.Properties.UeMtu
+				}
+
+				state.Identity, err = flattenMobileNetworkUserAssignedToNetworkLegacyIdentity(model.Identity)
+				if err != nil {
+					return fmt.Errorf("flattening `identity`: %+v", err)
+				}
+
+				properties := model.Properties
+
+				if properties.ControlPlaneAccessInterface.IPv4Address != nil {
+					state.ControlPlaneAccessIPv4Address = *properties.ControlPlaneAccessInterface.IPv4Address
+				}
```

Comment:
We don't need to nil check all of these, instead we should be using the pointer.From
```suggestion

				state.ControlPlaneAccessIPv4Address = pointer.From(properties.ControlPlaneAccessInterface.IPv4Address)
```